From 79fd5fcb2bb76d5f86e86df029aec655d0c7841c Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:08:27 -0400
Subject: [usb] sierra: remove redundant modem-control requests

Message-id: <1410887463-20674-436-git-send-email-dzickus@redhat.com>
Patchwork-id: 95262
O-Subject: [RHEL7 PATCH 435/591] USB: sierra: remove redundant modem-control requests
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit 7cdc3355a75584f35eab89ea324375594a61e81f
Author: Johan Hovold <jhovold@gmail.com>
Date:   Mon May 26 19:23:04 2014 +0200

    USB: sierra: remove redundant modem-control requests

    The tty-port implementation has already made sure that DTR/RTS have been
    raised and lowered by calling dtr_rts so remove the redundant calls from
    open and close.

    Signed-off-by: Johan Hovold <jhovold@gmail.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/serial/sierra.c b/drivers/usb/serial/sierra.c
index ac5e20d..3a2f5b8 100644
--- a/drivers/usb/serial/sierra.c
+++ b/drivers/usb/serial/sierra.c
@@ -769,14 +769,7 @@ static void sierra_close(struct usb_serial_port *port)
 
  portdata = usb_get_serial_port_data(port);
 
- portdata->rts_state = 0;
- portdata->dtr_state = 0;
-
- /* odd error handling due to pm counters */
- if (!usb_autopm_get_interface(serial->interface))
-  sierra_send_setup(port);
- else
-  usb_autopm_get_interface_no_resume(serial->interface);
+ usb_autopm_get_interface_no_resume(serial->interface);
 
  spin_lock_irq(&intfdata->susp_lock);
  portdata->opened = 0;
@@ -817,11 +810,6 @@ static int sierra_open(struct tty_struct *tty, struct usb_serial_port *port)
 
  portdata = usb_get_serial_port_data(port);
 
- /* Set some sane defaults */
- portdata->rts_state = 1;
- portdata->dtr_state = 1;
-
-
  endpoint = port->bulk_in_endpointAddress;
  for (i = 0; i < portdata->num_in_urbs; i++) {
   urb = sierra_setup_urb(serial, endpoint, USB_DIR_IN, port,
@@ -837,8 +825,6 @@ static int sierra_open(struct tty_struct *tty, struct usb_serial_port *port)
  if (err)
   goto err_submit;
 
- sierra_send_setup(port);
-
  spin_lock_irq(&intfdata->susp_lock);
  portdata->opened = 1;
  if (++intfdata->open_ports == 1)
-- 
1.7.1